### DISCUSSION TITLE
What is child priority?

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#1799] [General]
Hi,

I read about child.priority in the docs and have a question about the details. Perhaps this can help me with my setup.

I have a primary and a backup tunnel to a partner network. Obviously it would be bad if both tunnels are up and have a policy to route the traffic the their connection. I would like to have one tunnel up and one tunnel in standby.

On the other hand, perhaps child priority can help me here. Is is possible to configure the primary tunnel with a prio of 10 and the   backup tunnel with a prio of 20. Then the policy for the primary tunnel would kick in first, and all traffic would be encrypted according the the first tunnel. Only if that tunnel is down, the policy for the secondary tunnel would take effect.

Is my understanding of child priorities correct?

Michael.

**Comment by tobiasbrunner** [comment]
The priority is used to order the policies in the _kernel-netlink_ plugin and directly determines the priority of the policies in the Linux kernel. So if you have two CHILD_SAs with duplicate traffic selectors (so they get the same reqid) but different policies established concurrently, the policies for the one with the higher priority (numerically lower) will be installed in the kernel and therefore used for outbound traffic. Note that only one set of policies can be installed in the kernel as it doesn't allow duplicates and the priority isn't part of the key (so you won't see any policies with priority 20 as long as the other CHILD_SA  is installed).

> Only if that tunnel is down, the policy for the secondary tunnel would take effect.

If "down" means that the SA (in particular the CHILD_SA) is not established/installed, then yes.

**Comment by misch42** [comment]
Cool. sounds good. I will try.

**Comment by hariprar** [comment]
Hi,

I have a related question on the "priority" behavior. Let me know if it's better to open another thread than comment here.
In my case, I have policy1 and policy2 as shown below.
Is it possible for me to use the priority field to force which policy gets picked for an inbound connection from remote peer?
Is the priority field behavior different when the local host is initiator vs responder?

policy1
local: 192.168.1.1/32
remote: 192.168.2.1/32
priority: 10000000

policy2
local: 192.168.1.0/24
remote: 192.168.2.0/24
priority: 1

Thank you,
Hari

**Reply by tobiasbrunner** [reply]
> Is it possible for me to use the priority field to force which policy gets picked for an inbound connection from remote peer?

Not sure I understand what you mean. For an **inbound** connection? The priority is pretty much irrelevant for that (except maybe some performance impact when checking policies after decryption). For outbound traffic the priority is relevant, though, as it determines which SA is used to send a packet if it matches multiple policies.
